src/convertToObject.js

```diff
@@ -6,7 +6,38 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  // make an array with strings of sourceString
+  const properties = [];
+
+  let temp = sourceString.trim() + ';';
+
+  while (temp.includes(';')) {
+    properties.push(temp.slice(0, temp.indexOf(';') + 1));
+    temp = temp.slice(temp.indexOf(';') + 1).trim();
+  }
+
+  // make an object with properties
+  const propObj = {};
+
+  properties.forEach((str) => {
+    const prop = str.slice(0, str.indexOf(':')).trim();
+
+    const index = str.indexOf(';');
+
+    let val;
+
+    if (index !== -1) {
+      val = str.slice(str.indexOf(':') + 1, str.indexOf(';')).trim();
+    } else {
+      val = str.slice(str.indexOf(':') + 1).trim();
+    }
+
+    if (prop.length !== 0 && val.length !== 0) {
+      propObj[prop] = val;
+    }
+  });
```

Comment:
[CODE KNOWLEDGE] - Don't use iteration methods to modify an already existing array/object. (forEach is exception). These methods return a new array, so you should use it. In this case, you could directly reduce the array of properties into an object.